src/power_grid_model/data_types.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+Many data types are used throughout the power grid model project. In an attempt to clarify type hints, some types
+have been defined and explained in this file
+"""
+
+from typing import Any, Dict, List, Tuple, Union
+
+import numpy as np
+
+
+class SingleArray(np.ndarray):
```

Comment:
Subclassing `np.ndarray` is not that straightforward. See
https://numpy.org/doc/stable/user/basics.subclassing.html

I have doubts if we should do this. Or we wait until October this year. Then 3.11 is released, and 3.8 support will be dropped.

Reply:
I agree, but we're only using it for type hinting, so I guess we're fine then. I'll add some comments and I'll raise a `NotImplemented` exception in the `__new__()` method.

Reply:
It doesn't work :/ I've replaced SingleArray and DenseBatchArray simply with np.ndarray for now...